code/modules/visuals/render_steps.dm

```diff
@@ -69,3 +69,5 @@
 /atom/movable/render_step/emissive_blocker/Initialize(mapload, source)
 	. = ..()
 	src.color = GLOB.em_block_color
+	var/atom/movable/source_atom = locate(source)
```

Comment:
it would be better to just pass in the thing we're rendering off, rather then the source.
then we can read its render_target, and set our render_source to that in /render_step.

You would want to do this SET_PLANE_EXPLICIT at the /render_step level too, rather then here (also use initial(plane) to avoid double offsets)

While you're doing this work you should remove all instances of render_step's that sit in their parent's contents.
Just remember if they aren't immediately qdel'd after being created they'll need to delete when their parent deletes